Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.h

```diff
@@ -108,6 +108,13 @@ class Value {
         return *this;
     }
 
+    Value& operator=(const Value& value)
+    {
+        m_value = value.m_value;
+        m_type = value.m_type;
+        return *this;
```

Comment:
Do we generally not check for self assignment? I haven't noticed many classes in the codebase doing it.. Might be one of those old C++98 coding standard things.

Reply:
Does checking for self-assignment make sense when nothing can throw?
we could go over the codebase and see if anything suspicious pops up if there's a reason to worry about that?

in this specific case, both values are basically primitives anyhow, so I don't think anything can go wrong - do correct me if I'm wrong

Reply:
For this class, yeah, no issue. Other classes, it would only be required if 1) the self assignment operator was written poorly (e.g. delete [] m_data; m_data = new u8[other.m_size]; copy(m_data, other.m_data, other.m_size);), or 2) the assignment operator did a lot of expensive things that made it desirable to skip for self-assign.